uber/models.py

```diff
@@ -593,27 +593,12 @@ def auto_badge_num(self, badge_type):
 
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
-                first = 0
-                last = sametype.count() - 1
-                if sametype_list[last] - sametype_list[first] == last:
+                start, end = c.BADGE_RANGES[badge_type][0], sametype_list[-1]
+                gap_nums = sorted(set(range(start, end + 1)).difference(sametype_list))
```

Comment:
No, because we only use the badge range for the _first_ number - `end` is populated by the list of badges that we pulled from the DB. So if we have badges 3 through 100, with the badge range from 1 to 1000, then `start` is 1 but `end` is 100.

Originally both `start` and `end` were based on the badges from the DB, but in manual testing I found that you could create a permanent gap at the beginning of the range, hence the somewhat confusing use of the badge range here. There's also a corresponding unit test!